src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -26,6 +28,7 @@ sealed class MessagePump : IMessageReceiver, IDisposable
         readonly Action<string, Exception, CancellationToken> criticalErrorAction;
         readonly TimeSpan retryDelay;
         readonly string name;
+        readonly FastConcurrentLru<string, int> deliveryAttempts = new(1000);
```

Comment:
The change to removing the tight loop means that the only scenario where timeouts need to be tracked are where a single message takes longer than the acknowledgement timeout.

This means the cache size needs to be large enough to track those messages and not have them be evicted by other messages that are processing successfully.

I would expect that a count of 1000 is sufficient to cover that scenario except in the case where the message that needs to be retried is requeued and there are more than 1000 messages in the queue already.

Reply:
> I would expect that a count of 1000 is sufficient to cover that scenario except in the case where the message that needs to be retried is requeued and there are more than 1000 messages in the queue already.

The number of other messages in the queue should only matter for the case where the message is requeued to the end of the queue, but that only happens for quorum queues, which don't use the cached values in the first place.